text/0002-qri_dataset_definition.md

```diff
@@ -0,0 +1,343 @@
+| `entries`           | `int`         | number of top-level entries in the dataset. analogous to the number of rows in a table |
+| `format`            | `string`      | specifies the format of the raw data type by file extension. Must be one of: `json`|`csv`|`cbor` |
+| `formatConfig`      | `object`      |  removes as much ambiguity as possible about how to interpret the specified format. Properties of this object depend on the `format` field |
+| `length`            | `int`         | length of the data object in bytes |
+| `schema`            | `jsonSchema`  | the schema definition for the dataset body, schemas are defined using the IETF json-schema specification. for more info on json-schema see: https://json-schema.org |
+
+### `transform`
+Transform is a record of executing a transformation on data. Transforms can theoretically be anything from an SQL query, a jupyter
+notebook, the state of an ETL pipeline, etc, so long as the input is zero or more datasets, and the output is a single dataset
+Ideally, transforms should contain all the machine-necessary bits to deterministicly execute the algorithm referenced in "ScriptPath".
+
+_transform fields:_
+
+| name                 | type     | description |
+|----------------------|----------|-------------|
+| `scriptPath`         | `string` | the path to the script that produced this transformation |
+| `syntax`             | `string` | language this transform is written in. Only "skylark" is currently supported |
+| `syntaxVersion`      | `string` | an identifier for the application and version number that produced the result |
+| `config`             | `object` | any configuration that would affect the resulting hash. Transformations may use values present in config to perform their operations |
+| `resources`          | `object` |  map of all datasets transform depends on with both name and commit paths |
+
+### `viz`
+Viz stores configuration data related to representing a dataset as a visualization
+
+_viz fields:_
+
+| name         | type     | description |
+|--------------|----------|-------------|
+| `format`     | `string` | designates the visualization configuration syntax, currently only `html` is accepted |
+| `scriptPath` | `string` | location of script that generates the vizualization |
+
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+By defining datasets as a document, we run into the "package problem" of how
+other files a dataset depends on connect to this document. A dataset with a
+body that lists images leaves ambiguity around how it would be stored and 
+represented. Orienting around a document model does not alleviate the burden
+package managers take on more directly.
+
+#### Graph vs. File
+One of the key tensions is between _Graph_ and _File_ representations. By
+choosing to orient around a file metaphor instead of a database metaphor,
+it's harder to see how this document model has value in relation to DAG-based
+storage systems.
+
```

Comment:
Are you saying that 

"I understand that since we are in a DAG based storage system, it would seem more logical to orient around the metaphor of datasets as graphs rather than documents"?

If so, maybe we need to add explicitly somewhere why exactly we orient around a file rather than a graph. I understand that basically this entire rfc is an argument for exactly that, this section or a previous one might benefit from a little more clarity on that point.